### PR TITLE
Fix #289: add examples to travis (and fix example fallout from #285)

### LINKED ISSUE #289: Travis should attempt to build examples for the given platform.
Travis currently only builds and tests all the libraries. Linker errors, problems with link scripts, etc are all invisible to travis builds currently. Travis should build the examples for the current platform.

I didn't implement this originally because I"m not that experienced with travis and couldn't think of a good way to have it only build *only* the examples for the **current platform** at the time when I rewrote the travis script for the new build system.

### PR DESCRIPTION
**Pull request by mcoffin**
* Fixes #289
* Fixes some issues with some examples that were introduced when #285 landed.

### CODE REVIEW

File: src/hal/k20/layout.ld [hunk 1]
```diff
@@ -1,3 +1,7 @@
+__aeabi_unwind_cpp_pr0 = abort;
+__aeabi_unwind_cpp_pr1 = abort;
+__aeabi_unwind_cpp_pr2 = abort;
+
```
Comment (mcoffin) [inline src/hal/k20/layout.ld]: This was the invisible fallout on the k20 platform created by #285. It didn't show up until we started compiling the examples in travis builds.

File: ci/build-examples.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+#
+# Zinc, the bare metal stack for rust.
+# Copyright 2014 Vladimir "farcaller" Pouzanov <farcaller@gmail.com>
```
Comment (farcaller) [inline ci/build-examples.sh]: Fix author name.

**Comment by farcaller** [comment]
A minor comment notice, otherwise LGTM

**Comment by mcoffin** [comment]
@farcaller Changed author in header. My bad I've just been pasting it.

**Pull request merged**